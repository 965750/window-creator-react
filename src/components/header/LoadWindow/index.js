import { connect } from 'react-redux'
import { faArrowAltCircleDown, faTrash } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { loadWindow, removeWindowSave } from '../../../store/actions/creatorActions'

import moment from 'moment'
import PropTypes from 'prop-types'
import React from 'react'
import styles from './LoadWindow.module.scss'

const LoadWindow = ({
  window, isLast, loadWindow, removeWindowSave, // eslint-disable-line
}) => (
  <div data-testid="wrapper" className={`${styles.save} ${isLast ? '' : 'border-b border-silver'} py-2 text-left flex relative`}>
    <div>
      <p className={`${styles.save__name}`}>{window.save.name}</p>
      <p className={`${styles.save__time} text-gullGray opacity-75`}>{moment(window.save.time).fromNow()}</p>
    </div>
    <FontAwesomeIcon
      data-testid="loadIcon"
      onClick={() => loadWindow(window.id)}
      className={`${styles.save__btnLoad} text-hippieBlue absolute top-half right-30`}
      icon={faArrowAltCircleDown}
    />
    <FontAwesomeIcon
      data-testid="removeIcon"
      onClick={() => removeWindowSave(window.id)}
      className={`${styles.save__btnDelete} text-error absolute top-half right-0`}
      icon={faTrash}
    />
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  loadWindow: (id) => dispatch(loadWindow(id)),
  removeWindowSave: (id) => dispatch(removeWindowSave(id)),
})

LoadWindow.propTypes = {
  loadWindow: PropTypes.func,
  removeWindowSave: PropTypes.func,
  isLast: PropTypes.bool,
  window: PropTypes.shape({
    save: PropTypes.shape({
      name: PropTypes.string,
      time: PropTypes.number,
    }),
    id: PropTypes.string,
  }),
}

export default connect(null, mapDispatchToProps)(LoadWindow);
