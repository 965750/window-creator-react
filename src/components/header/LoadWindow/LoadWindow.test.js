import '@testing-library/jest-dom'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup, fireEvent } from '@testing-library/react'

import LoadWindow from './index'
import messages from '../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
  firebase: {
    profile: {
      windows: [{
        id: 'idMock'
      }],
    },
    auth: {
      uid: 'uidMock',
    }
  },
  notification: null
}
const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('LoadWindow', () => {
  let wrapper
  const initialProps = {
    window: {
      save: {
        name: 'mockName',
        time: 123
      },
      id: 'idMock'
    },
    loadWindow: jest.fn(),
    isLast: true,
    removeWindowSave: jest.fn(),
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <LoadWindow {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should have border classes on wrapper', () => {
    setUp({...initialProps, isLast: false})

    const wrapperDiv = wrapper.getByTestId('wrapper')

    expect(wrapperDiv.classList.contains('border-b')).toBe(true)
  })
})
