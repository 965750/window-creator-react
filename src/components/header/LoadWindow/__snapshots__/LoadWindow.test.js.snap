// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadWindow should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="save  py-2 text-left flex relative"
    data-testid="wrapper"
  >
    <div>
      <p
        class="save__name"
      >
        mockName
      </p>
      <p
        class="save__time text-gullGray opacity-75"
      >
        50 years ago
      </p>
    </div>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-arrow-alt-circle-down fa-w-16 save__btnLoad text-hippieBlue absolute top-half right-30"
      data-icon="arrow-alt-circle-down"
      data-prefix="fas"
      data-testid="loadIcon"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zM212 140v116h-70.9c-10.7 0-16.1 13-8.5 20.5l114.9 114.3c4.7 4.7 12.2 4.7 16.9 0l114.9-114.3c7.6-7.6 2.2-20.5-8.5-20.5H300V140c0-6.6-5.4-12-12-12h-64c-6.6 0-12 5.4-12 12z"
        fill="currentColor"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-trash fa-w-14 save__btnDelete text-error absolute top-half right-0"
      data-icon="trash"
      data-prefix="fas"
      data-testid="removeIcon"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
        fill="currentColor"
      />
    </svg>
  </div>
</DocumentFragment>
`;
